golang/x/relay/keeper/chain.go

```diff
@@ -92,19 +93,25 @@ func (k Keeper) IsMostRecentCommonAncestor(ctx sdk.Context, ancestor, left, righ
 func (k Keeper) HeaviestFromAncestor(ctx sdk.Context, ancestor, currentBest, newBest types.Hash256Digest, limit uint32) (types.Hash256Digest, sdk.Error) {
 	ancestorBlock, err := k.GetHeader(ctx, ancestor)
 	if err != nil {
-		return types.Hash256Digest{}, types.ErrUnknownBlock(types.DefaultCodespace)
+		return types.Hash256Digest{}, types.ErrUnknownBlock(types.DefaultCodespace, "ancestor", ancestor[:])
 	}
 	leftBlock, err := k.GetHeader(ctx, currentBest)
 	if err != nil {
-		return types.Hash256Digest{}, types.ErrUnknownBlock(types.DefaultCodespace)
+		return types.Hash256Digest{}, types.ErrUnknownBlock(types.DefaultCodespace, "currentBest", currentBest[:])
 	}
 	rightBlock, err := k.GetHeader(ctx, newBest)
 	if err != nil {
-		return types.Hash256Digest{}, types.ErrUnknownBlock(types.DefaultCodespace)
+		return types.Hash256Digest{}, types.ErrUnknownBlock(types.DefaultCodespace, "newBest", newBest[:])
 	}
 
-	if leftBlock.Height < ancestorBlock.Height || rightBlock.Height < ancestorBlock.Height {
-		return types.Hash256Digest{}, types.ErrBadHeight(types.DefaultCodespace)
+	if leftBlock.Height < ancestorBlock.Height {
+		errDetails := fmt.Sprintf("currentBest %x is below the ancestor height", currentBest)
```

Comment:
big string here bc `ErrBadHeight` caters to 2 different situations. was thinking of making 2 separate ones. the other instance is at diff-0269e6a4128706217e8b62b4cf2eeed3R151

Reply:
oh yeah, it might be better to split those into 2 separate errors